File: frontend/lib/models/vendor.dart
class Vendor {
  final String id;
  final String userId;
  final String storeName;
  final String? storeDescription;
  final String? storeLogoUrl;
  final String? storeBannerUrl;
  final String address;
  final String city;
  final String state;
  final double deliveryRadiusKm;
  final String status;
  final bool isActive;
  final double rating;
  final int totalOrders;
  final DateTime createdAt;

  Vendor({
    required this.id,
    required this.userId,
    required this.storeName,
    this.storeDescription,
    this.storeLogoUrl,
    this.storeBannerUrl,
    required this.address,
    required this.city,
    required this.state,
    this.deliveryRadiusKm = 10.0,
    required this.status,
    this.isActive = true,
    this.rating = 0.0,
    this.totalOrders = 0,
    required this.createdAt,
  });

  factory Vendor.fromJson(Map<String, dynamic> json) {
    return Vendor(
      id: json['id'],
      userId: json['user_id'],
      storeName: json['store_name'],
      storeDescription: json['store_description'],
      storeLogoUrl: json['store_logo_url'],
      storeBannerUrl: json['store_banner_url'],
      address: json['address'],
      city: json['city'],
      state: json['state'],
      deliveryRadiusKm:
          (json['delivery_radius_km'] as num?)?.toDouble() ?? 10.0,
      status: json['status'] ?? 'pending',
      isActive: json['is_active'] ?? true,
      rating: (json['rating'] as num?)?.toDouble() ?? 0.0,
      totalOrders: json['total_orders'] ?? 0,
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}

File: backend/app/schemas/vendor.py
"""Vendor schemas."""
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID
from datetime import datetime, time
from app.models.vendor import VendorStatus


class VendorRegisterRequest(BaseModel):
    store_name: str = Field(..., min_length=2, max_length=200)
    store_description: Optional[str] = None
    address: str
    city: str
    state: str
    postal_code: str
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    delivery_radius_km: float = 10.0
    gstin: Optional[str] = None
    pan_number: Optional[str] = None
    fssai_license: Optional[str] = None
    bank_account_number: Optional[str] = None
    bank_ifsc: Optional[str] = None
    bank_name: Optional[str] = None


class VendorUpdate(BaseModel):
    store_name: Optional[str] = Field(None, min_length=2, max_length=200)
    store_description: Optional[str] = None
    store_logo_url: Optional[str] = None
    store_banner_url: Optional[str] = None
    address: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    postal_code: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    delivery_radius_km: Optional[float] = None


class VendorResponse(BaseModel):
    id: UUID
    user_id: UUID
    store_name: str
    store_description: Optional[str] = None
    store_logo_url: Optional[str] = None
    store_banner_url: Optional[str] = None
    address: str
    city: str
    state: str
    postal_code: str
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    delivery_radius_km: float
    commission_rate: float
    status: VendorStatus
    is_active: bool
    rating: float
    total_orders: int
    created_at: datetime

    class Config:
        from_attributes = True


class VendorAdminUpdate(BaseModel):
    status: Optional[VendorStatus] = None
    commission_rate: Optional[float] = None
    is_active: Optional[bool] = None


class StoreTimingsCreate(BaseModel):
    day_of_week: int = Field(..., ge=0, le=6)
    open_time: time
    close_time: time
    is_closed: bool = False


class StoreTimingsResponse(BaseModel):
    id: UUID
    day_of_week: int
    open_time: time
    close_time: time
    is_closed: bool

    class Config:
        from_attributes = True


class VendorDashboardStats(BaseModel):
    total_orders: int = 0
    pending_orders: int = 0
    total_revenue: float = 0.0
    total_products: int = 0
    avg_rating: float = 0.0
    daily_orders: list = []
    top_products: list = []
